In [1]:
!pip install ffmpeg

In [2]:
#! işareti: bir komut satırı komutunu çalıştırmak için kullanılır. 
#!pip install ffmpeg komutunu kullanarak, Python ortamının dışında, komut satırında ffmpeg adlı bir yazılımı kurmayı veya 
#güncellemeyi deniyoruz.
#pip python paket yöneticisi
#ffmpeg: coklu ortam dosyası biçimlerini dönüştürmek, ses ve video dosyalarını işlemek için kullanılan 
#açık kaynaklı bir çoklu ortam yazılımıdır. 
#ses veya video işleme yaparken ffmpeg gibi araçları kullanmanız gerekebilir.
#Python paket yöneticisi pip aracılığıyla ffmpeg'yi kurmaya çalışıyorsunuz.

In [3]:
#conda install -c conda-forge ffmpeg

In [4]:
#Conda paket yöneticisi kullanılarak FFmpeg çoklu ortam çerçevesini kuruyoruz.
#ffmpeg kurmak istediğimiz paketin adı
#-c conda-forge: bu kanal araciligiyla ffmpeg paketini kurmaya calisiyoruz

In [5]:
!pip install bar_chart_race

In [6]:
#bar_chart_race - python'da bar_chart_race adli ucuncu taraf bir kutuphaneyi kurmak icin yukaridaki kodu yazdik
#bu kutuphane cubuk grafik yarisi olarak bilinen veri gorsellestirmesi turunu olusturmak icin kullanislidir.
#zaman içindeki veri değişimini göstermek için kullanılır.

In [7]:
import pandas as pd
import matplotlib.pyplot as plt

%matplotlib inline
import seaborn as sns
import bar_chart_race as bcr

In [8]:
#import matplotlib.pyplot as plt: matplotlib kütüphanesini içeri aktarıyoruz. bu kütüphane çeşitli grafik türlerini oluşturmak
#ve bu grafikleri özelleştirmek için kullanılan güçlü bir grafik çizme kütüphanesidir.
#Python'da veri görselleştirmesi yapmak için yaygın olarak matplotlib kütüphanesi kullanılır.

In [9]:
#%matplotlib inline: bu komut Matplotlib kütüphanesini kullanarak oluşturulan grafikleri Jupyter arayüzünde görüntülemek için
#kullanılır.

In [46]:
df=pd.read_csv('corona_dat.csv')

In [47]:
df.sample()   #veri setinden bir ornege bakariz

date  Afghanistan  Albania  Algeria  Andorra  Angola  \
61  3/23/2020            0       17       29       20       1   

    Antigua and Barbuda  Argentina  Armenia  Australia  ...  Uruguay     US  \
61                    2         35       41        133  ...        4  10767   

    Uzbekistan  Venezuela  Vietnam  West Bank and Gaza  Western Sahara  Yemen  \
61           3          7       10                   7               0      0   

    Zambia  Zimbabwe  
61       0         1  

[1 rows x 188 columns]

In [48]:
df.shape

(112, 188)

In [49]:
list(df.columns)   #veri setimizdeki butun kolonlari listeledik

['date',
 'Afghanistan',
 'Albania',
 'Algeria',
 'Andorra',
 'Angola',
 'Antigua and Barbuda',
 'Argentina',
 'Armenia',
 'Australia',
 'Austria',
 'Azerbaijan',
 'Bahamas',
 'Bahrain',
 'Bangladesh',
 'Barbados',
 'Belarus',
 'Belgium',
 'Belize',
 'Benin',
 'Bhutan',
 'Bolivia',
 'Bosnia and Herzegovina',
 'Botswana',
 'Brazil',
 'Brunei',
 'Bulgaria',
 'Burkina Faso',
 'Burma',
 'Burundi',
 'Cabo Verde',
 'Cambodia',
 'Cameroon',
 'Canada',
 'Central African Republic',
 'Chad',
 'Chile',
 'China',
 'Colombia',
 'Comoros',
 'Congo (Brazzaville)',
 'Congo (Kinshasa)',
 'Costa Rica',
 "Cote d'Ivoire",
 'Croatia',
 'Cuba',
 'Cyprus',
 'Czechia',
 'Denmark',
 'Diamond Princess',
 'Djibouti',
 'Dominica',
 'Dominican Republic',
 'Ecuador',
 'Egypt',
 'El Salvador',
 'Equatorial Guinea',
 'Eritrea',
 'Estonia',
 'Eswatini',
 'Ethiopia',
 'Fiji',
 'Finland',
 'France',
 'Gabon',
 'Gambia',
 'Georgia',
 'Germany',
 'Ghana',
 'Greece',
 'Grenada',
 'Guatemala',
 'Guinea',
 'Guinea-Bissau',
 

In [50]:
df2 = df[['date', 'China', 'Italy', 'Spain', 'Turkey', 'US', 'United Kingdom', 'Sweden', 'Germany', 'Finland','France', 'Egypt', 'Netherlands', 'Portugal', 'Russia', 'Ukraine', 'New Zealand', 'Israel', 'Hungary', 'Croatia',]]
#tarih degiskenimizi tutup gormek istedigimiz kolonlari seciyoruz. df2 degiskenimize atiyoruz

In [51]:
df2.head()

date  China  Italy  Spain  Turkey  US  United Kingdom  Sweden  \
0  1/22/2020    593      0      0       0   1               0       0   
1  1/23/2020     98      0      0       0   0               0       0   
2  1/24/2020    291      0      0       0   1               0       0   
3  1/25/2020    505      0      0       0   0               0       0   
4  1/26/2020    693      0      0       0   3               0       0   

   Germany  Finland  France  Egypt  Netherlands  Portugal  Russia  Ukraine  \
0        0        0       0      0            0         0       0        0   
1        0        0       0      0            0         0       0        0   
2        0        0       2      0            0         0       0        0   
3        0        0       1      0            0         0       0        0   
4        0        0       0      0            0         0       0        0   

   New Zealand  Israel  Hungary  Croatia  
0            0       0        0        0  
1            0       0        0        0  
2            0       0        0        0  
3            0       0        0        0  
4            0       0        0        0

In [52]:
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 112 entries, 0 to 111
Data columns (total 20 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   date            112 non-null    object
 1   China           112 non-null    int64 
 2   Italy           112 non-null    int64 
 3   Spain           112 non-null    int64 
 4   Turkey          112 non-null    int64 
 5   US              112 non-null    int64 
 6   United Kingdom  112 non-null    int64 
 7   Sweden          112 non-null    int64 
 8   Germany         112 non-null    int64 
 9   Finland         112 non-null    int64 
 10  France          112 non-null    int64 
 11  Egypt           112 non-null    int64 
 12  Netherlands     112 non-null    int64 
 13  Portugal        112 non-null    int64 
 14  Russia          112 non-null    int64 
 15  Ukraine         112 non-null    int64 
 16  New Zealand     112 non-null    int64 
 17  Israel          112 non-null    int64 
 18  Hungary   

In [53]:
df2.set_index("date", inplace=True)
#date'r gore yani tarihe gore ulkelerin korona verilerine baktik. set index ile veriyi index olarak ayarladik.
#yani date kolonunu indeks yani 0 1 2 3 4 satiri haline getirdik.
#inplace true parametresi ile de yapilan degisiklikleri kalici hale getirdik.

In [54]:
df2.head()

China  Italy  Spain  Turkey  US  United Kingdom  Sweden  Germany  \
date                                                                          
1/22/2020    593      0      0       0   1               0       0        0   
1/23/2020     98      0      0       0   0               0       0        0   
1/24/2020    291      0      0       0   1               0       0        0   
1/25/2020    505      0      0       0   0               0       0        0   
1/26/2020    693      0      0       0   3               0       0        0   

           Finland  France  Egypt  Netherlands  Portugal  Russia  Ukraine  \
date                                                                        
1/22/2020        0       0      0            0         0       0        0   
1/23/2020        0       0      0            0         0       0        0   
1/24/2020        0       2      0            0         0       0        0   
1/25/2020        0       1      0            0         0       0        0   
1/26/2020        0       0      0            0         0       0        0   

           New Zealand  Israel  Hungary  Croatia  
date                                              
1/22/2020            0       0        0        0  
1/23/2020            0       0        0        0  
1/24/2020            0       0        0        0  
1/25/2020            0       0        0        0  
1/26/2020            0       0        0        0

In [43]:
df_total = df2.cumsum(axis=0)  #kumulatif veri

#cumsum: kumulatif toplamlari hesaplamak icin kullanilir. axis=0 satirdaki axia=1 sutundaki
#amacimiz her ulkenin korona verilerini her ulke bazinda kamulatif toplayip diger ulkelerle karsilastirmak
#df_total degisknine atayip ana verimizi keybetmeden bu degisken uzerinden devam ettik
#df_total bizimyeni veri cercevemiz oldu. bu veri cercevesinin her bir sutunu ilgili sutunun kumulatif toplamini icerir.

verimiz bu:
1/22/2020	593
1/23/2020	98
1/24/2020	291
1/25/2020	505
1/26/2020	693

kumulatiof toplamda:
1/22/2020	593
1/23/2020	691
1/24/2020	982
1/25/2020	1487
1/26/2020	2180

en sonunda cikti alirken grafikte surekli artisin olmasi icin bunu yapiyoruz.
yani toplanarak gidiyor butun verilerimiz.

In [20]:
bcr.bar_chart_race(df=df_total, filename="covid19.mp4", figsize=(8,6), title="Bazi Ulkelerdeki Covid-19 Vaka Sayilari Grafigi")

In [45]:
df2.head()

China  Italy  Spain  Turkey  US  United Kingdom  Sweden  Germany  \
date                                                                          
1/22/2020    593      0      0       0   1               0       0        0   
1/23/2020     98      0      0       0   0               0       0        0   
1/24/2020    291      0      0       0   1               0       0        0   
1/25/2020    505      0      0       0   0               0       0        0   
1/26/2020    693      0      0       0   3               0       0        0   

           Finland  France  Egypt  Netherlands  Portugal  Russia  Ukraine  \
date                                                                        
1/22/2020        0       0      0            0         0       0        0   
1/23/2020        0       0      0            0         0       0        0   
1/24/2020        0       2      0            0         0       0        0   
1/25/2020        0       1      0            0         0       0        0   
1/26/2020        0       0      0            0         0       0        0   

           New Zealand  Israel  Hungary  Croatia  
date                                              
1/22/2020            0       0        0        0  
1/23/2020            0       0        0        0  
1/24/2020            0       0        0        0  
1/25/2020            0       0        0        0  
1/26/2020            0       0        0        0